torch/utils/_sympy/functions.py

```diff
@@ -247,17 +279,122 @@ def eval(cls, p, q):
         if sympy.Mod(p, q) == 0:
             return S.Zero
 
-    def _eval_is_integer(self):
-        p, q = self.args
-        return fuzzy_and([p.is_integer, q.is_integer, fuzzy_not(q.is_zero)])  # type: ignore[attr-defined]
-
+    # NB: args[1] for PythonMod
     def _eval_is_nonnegative(self):
         return True if self.args[1].is_positive else None  # type: ignore[attr-defined]
 
     def _eval_is_nonpositive(self):
         return True if self.args[1].is_negative else None  # type: ignore[attr-defined]
 
 
+# C-style modulus: take sign from LHS
+class CMod(sympy.Function):
+    nargs = (2,)
+
+    is_integer = True
+
+    @classmethod
+    def eval(cls, p, q):
+        assert p.is_integer, p
+        assert q.is_integer, q
+
+        if q.is_zero:
+            raise ZeroDivisionError("Modulo by zero")
+
+        # Three cases:
+        #   1. p == 0
+        #   2. p is either q or -q
+        #   3. p is integer and q == 1
+        if p is S.Zero or p in (q, -q) or q == 1:
+            return S.Zero
+
+        # Evaluate if they are both literals.
+        if q.is_Number and p.is_Number:
+            r = p % q
+            # C modulus behavior!
+            if p < 0:
+                r *= -1
```

Comment:
This is not correct? you should do `r -= q` or, even better, `math.remainder`. Plz add test.

Reply:
You are right. CMod not used by anything right now, probably we can just dump it for now

Reply:
It should probably be used to lower `torch.remainder`